### ISSUE TITLE
Error: Failed to join cluster: Failed request to add member: The joining server version doesn't match (expected 5.0.3 with API count 340)

### ISSUE THREAD

**Issue by aanderse** [canonical/lxd#14575]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 24.04.1 LTS
 * The output of "snap list --all lxd core20 core22 core24 snapd":
```
Name    Version        Rev    Tracking       Publisher    Notes
core20  20240705       2379   latest/stable  canonical**  base
lxd     5.0.3-80aeff7  29351  5.0/stable     canonical**  -
snapd   2.63           21759  latest/stable  canonical**  snapd
```
 * The output of "lxc info" or if that fails:
   * Kernel version: `6.8.0-45-generic`snap 
   * LXC version: `5.0.3` (presumably the same as `LXD version` below
   * LXD version: ` 5.0.3-80aeff7`
   * Storage backend in use: `zfs`

# Issue description

i have a cluster of `ubuntu` servers running `lxd` - all of the servers are running `lxd` version `5.0.3-*`. i am now trying to add a **new** server to the cluster, but `snap` is no longer offering `lxd` version `5.0.3-*`... this shouldn't be a problem i thought because it is a LTS version... but when i try to add the new server to the cluster i get this error:

```
Error: Failed to join cluster: Failed request to add member: The joining server version doesn't match (expected 5.0.3 with API count 340)
```

does this mean that `lxd` LTS has introduced an incompatibility? **how can i use `lxd` cli to check what my API count is?**

# Steps to reproduce

since `snap` isn't listing `lxd` version `5.0.3-*` anymore you won't be able to reproduce this as you can't get the older LTS revision

# Information to attach

 - [ ] Any relevant kernel output (`dmesg`)
 - [ ] Container log (`lxc info NAME --show-log`)
 - [ ] Container configuration (`lxc config show NAME --expanded`)
 - [ ] Main daemon log (at /var/log/lxd/lxd.log or /var/snap/lxd/common/lxd/logs/lxd.log)
 - [ ] Output of the client with --debug
 - [ ] Output of the daemon with --debug (alternatively output of `lxc monitor` while reproducing the issue)

---

thank you very much for your consideration and any help you can provide :bowing_man:

**Comment by tomponline** [comment]
When running a LXD cluster all cluster members need to be the same version.

We recently released LXD 5.0.4, so I suspect your new host is getting this one.

https://discourse.ubuntu.com/t/lxd-5-0-4-lts-has-been-released/49681

So would suggest upgrading your other members first to LXD 5.0.4 before adding a new member.

The LTS version commitment does not allow mixing sub-versions between cluster members, but rather than we commit not to introduce DB schema changes or other API breakages that would prevent downgrading back to a previous snap revision.

If you are needing the ability to pin a deployment at a certain revision cluster wide then snap provides some options for this, please see:

https://documentation.ubuntu.com/lxd/en/latest/howto/snap/#use-a-snap-store-proxy

https://documentation.ubuntu.com/lxd/en/latest/howto/cluster_manage/#upgrade-cluster-members

There's also snap cohorts, but that may not be as useful in this case:

https://discourse.ubuntu.com/t/managing-the-lxd-snap-package/37214#cohorts-pinning

**Comment by tomponline** [comment]
As you can get the snap revision from the other cluster members, you can also try doing `snap install lxd --revision=` to see if the snap store allows you to install the specific revision for LXD 5.0.3.

**Comment by tomponline** [comment]
For future reference we have a support forum for general questions/problems https://discourse.ubuntu.com/c/lxd/support/149

**Comment by aanderse** [comment]
> When running a LXD cluster all cluster members need to be the same version.

i did not understand that even minor version is relevant here... it seems like an unfortunate policy for ubuntu to start purging versions from `snap` then

> As you can get the snap revision from the other cluster members, you can also try doing `snap install lxd --revision=` to install the specific revision for LXD 5.0.3.

right, i tried that but the version was not available unfortunately - i appreciate you confirming what i tried, though

> We recently released LXD 5.0.4, so I suspect your new host is getting this one.

yes, exactly this


---


thank you for your notes, i find them helpful and i learned a few important things i did not know before :bowing_man:

**Comment by tomponline** [comment]
You can get your revision here (i tried it) https://github.com/canonical/lxd/issues/14575#issuecomment-2514543763
